src/backend/launcher.ts

```diff
@@ -826,13 +826,16 @@ export async function verifyWinePrefix(
 
   return command
     .then((result) => {
-      // This is kinda hacky
-      const wasUpdated = result.stderr.includes(
-        wineVersion.type === 'proton'
-          ? 'Proton: Upgrading prefix from'
-          : 'has been updated'
-      )
-      return { res: result, updated: wasUpdated }
+      const currentWinePath = join(winePrefix, 'current_wine')
+      if (
+        !existsSync(currentWinePath) ||
+        readFileSync(currentWinePath, 'utf-8') != wineVersion.bin
+      ) {
+        writeFileSync(currentWinePath, wineVersion.bin, 'utf-8')
+        return { res: result, updated: true }
+      } else {
+        return { res: result, updated: false }
```

Comment:
instead of making this update, could you make this "new"? This would create an empty file every time heroic runs first time setup and then would prevent it from rerunning the setup unnecessarily on prefix updates